### PR TITLE
User local config

### PR DESCRIPTION
**Pull request by billw2012**
Splits debug config into a general and a user part. 
I tried to make git ignore changes to the user part with .gitignore, but GK still shows changes. Very annoying :/

### CODE REVIEW

File: Vindicta.Altis/config/global_config.hpp [hunk 1]
```diff
@@ -1,48 +1,15 @@
...
 
-// Default oop config overrides.
-// Nothing here cos we don't want to override any stuff by default!
-// Undefine debug and info logging, leave warning and error logging.
-//#undef OOP_DEBUG
-//#undef OOP_INFO
-//#undef OOP_WARNING
-//#undef OOP_ERROR
+// Various runtime OOP assertions (class existence, member variable existence, etc)
+#define OOP_ASSERT
```
Comment (zalexki) [inline Vindicta.Altis/config/global_config.hpp]: @billw2012 so now the assert is by default ? looks like it was undef previously

Comment (zalexki) [inline Vindicta.Altis/config/global_config.hpp]: does this add checks and lower perf ? maybe we should define it to undef by default to ease release build ?

Comment (billw2012) [reply Vindicta.Altis/config/global_config.hpp]: This applies only to debug build, release uses `config/release/global_config.hpp`. Although looking now we have 3 global_config.hpp, configs\dev configs\release and Vindicta.Altis. Not sure anymore which are used when.

Comment (billw2012) [reply Vindicta.Altis/config/global_config.hpp]: Actually I guess I remember now, it uses the Vindicta.Altis one, and copies over it for making the pbos. So this change is okay I guess? We want asserts on or off by default when played through editor? Maybe off?

Comment (zalexki) [reply Vindicta.Altis/config/global_config.hpp]: hum ye ok i see well then maybe it make sens to define more debug thing by default if there is a release_config file

**Comment by Sparker95** [comment]
I think it's a pretty good solution!

**Pull request merged**